### PR TITLE
[refactor] Move configuration files to single YAML file

### PR DESCRIPTION
**Pull request by ervteng**
### Proposed change(s)

This PR refactors the YAML configuration files into a per-environment format. Each YAML now contains both the behavior configurations/hyperparameters, as well as curriculum and parameter variation configurations. The example YAMLs were reformatted to reflect this. 

To be done in this PR:

- ~Documentation changes~

Not done in this PR:

- Move engine configurations and other CLI options (num_envs, env, seed) into YAML files as well
- Save out final YAML file. 

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)
https://docs.google.com/document/d/1uyk5JVNevfWy2DmqVHy19lcDzyQCNppPxkPOgJcQ7z4/edit#heading=h.sozm3sblhtke

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [x] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

**Comment by ervteng** [comment]
@anupambhatnagar - this will definitely break daily CI - what steps should I take to adapt the ml-agents-cloud utility to match?

File: com.unity.ml-agents/CHANGELOG.md [hunk 1]
```diff
@@ -36,7 +36,10 @@ and this project adheres to
   communication between Unity and the Python process.
 - The obsolete `Agent` methods `GiveModel`, `Done`, `InitializeAgent`,
   `AgentAction` and `AgentReset` have been removed.
-- The GhostTrainer has been extended to support asymmetric games and the asymmetric example environment Strikers Vs. Goalie has been added.
+- The GhostTrainer has been extended to support asymmetric games and the asymmetric example environment Strikers Vs.   Goalie has been added.
+- Curriculum and Parameter Randomization configurations have been merged
+  into the main training configuration file. Note that this means training
+  configuration files are environment-specific. (#3791)
```
Comment (awjuliani) [inline com.unity.ml-agents/CHANGELOG.md:42]:
```suggestion
  configuration files are now environment-specific. (#3791)
```

**Comment by ervteng** [comment]
@awjuliani I still want to split the YAML to trainer and model. With luck that will be done after the other CLI and output file changes are done. 

The files were generated using a script that loaded the old config's default and overrode them using the config, then outputted the new YAML. I'm also worried about the files diverging before we are able to create the default values in code. This PR is technically still compatible with the old config files (as long as you add `behaviors:`) to it. Would it be a good idea to have an intermediary step that uses the old `trainer_config` and only migrates generalization/curriculum? We

File: config/ppo/StrikerVsGoalie.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    buffer_size: 20480
+    hidden_units: 512
+    time_horizon: 1000
+    num_layers: 2
+    self_play:
+      window: 10
+      play_against_latest_model_ratio: 0.5
+      save_steps: 50000
+      swap_steps: 100000
+      team_change: 200000
```
Comment (mmattar) [inline config/ppo/StrikerVsGoalie.yaml:32]: add a new line

File: config/ppo/WallJump.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    time_horizon: 128
+    sequence_length: 64
+    summary_freq: 20000
+    use_recurrent: false
+    vis_encode_type: simple
+    reward_signals:
+      extrinsic:
+        strength: 1.0
+        gamma: 0.99
+
```
Comment (mmattar) [inline config/ppo/WallJump.yaml:51]: remove unnecessary line

File: config/ppo/StrikerVsGoalie.yaml (hunk 2 above)
Comment (ervteng) [reply config/ppo/StrikerVsGoalie.yaml]: Done

File: config/ppo/WallJump.yaml (hunk 3 above)
Comment (ervteng) [reply config/ppo/WallJump.yaml]: Done

File: config/ppo/3DBall_randomize.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        min_value: 0.5
+        max_value: 10
+    gravity:
+        sampler-type: "uniform"
+        min_value: 7
+        max_value: 12
+    scale:
+        sampler-type: "uniform"
+        min_value: 0.75
+        max_value: 3
```
[the remark this answers is not in the document]
Comment (ervteng) [reply config/ppo/3DBall_randomize.yaml]: Done

File: com.unity.ml-agents/CHANGELOG.md (hunk 1 above)
Comment (ervteng) [reply com.unity.ml-agents/CHANGELOG.md]: Done

**Pull request merged**